### PR TITLE
Turn on more sass-lint rules, add autoprefixer

### LINKED ISSUE #1764: update sass-lint config
I'd like to turn on a few more rules that will enforce better CSS practices.

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

closes #1764 

#### What's this PR do?

This turns on a bunch more sass-lint rules. I also updated sass-lint and installed the `postcss` loader for webpack, with the [autoprefixer](https://github.com/postcss/autoprefixer) plugin (which automatically adds vendor prefixes).

#### How should this be manually tested?

Basically, make sure adding the sass-lint rules makes sense, and also that I didn't clobber anything on the site in the process.

### CODE REVIEW

File: static/scss/footer.scss [hunk 1]
```diff
@@ -22,9 +22,9 @@
     margin: 24px 0;
 
     a {
-      display:inline-block;
-      margin:0 24px 0 0;
-      color:#4c8399;
+      display: inline-block;
+      margin: 0 24px 0 0;
+      color: #5c8399;
```
Comment (noisecapella) [inline static/scss/footer.scss]: Is the color change intentional here?

File: postcss.config.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  plugins: [
+    require('autoprefixer')({
+      browsers: ['> 5%']
```
Comment (noisecapella) [inline postcss.config.js]: When I run the command line tool to list these browsers I only get these four:

    and_chr 53
    and_uc 11
    chrome 53
    ios_saf 10

If we change to 1 % we get this list instead:

    and_chr 53
    and_uc 11
    android 4.4.3-4.4.4
    android 4.4
    chrome 54
    chrome 53
    chrome 49
    edge 14
    firefox 49
    ie 11
    ios_saf 10
    ios_saf 9.3
    op_mini all
    safari 10
    samsung 4

**Comment by noisecapella** [comment]
There are a few places in `homepage.scss` which still use the prefixes, can you remove the extraneous ones there as well?

File: postcss.config.js (hunk 2 above)
Comment (aliceriot) [inline postcss.config.js]: cool, I meant to leave a comment asking what value would be good for this setting. > 1% it is!

File: static/scss/footer.scss (hunk 1 above)
Comment (aliceriot) [inline static/scss/footer.scss]: oh, no, must have been too quick when editing.

**Comment by aliceriot** [comment]
Oh yeah, I'll see how difficult it is to remove `homepage.scss` from the ignore list too.

**Comment by aliceriot** [comment]
I removed `homepage.scss` from the ignore list, deleted a bunch of unused CSS from it, and then fixed all the errors in it.

Also noticed I had goofed up on the financial aid review page, so I fixed that real quick as well.

**Pull request merged**